libs/internal/include/launchdarkly/network/asio_requester.hpp

```diff
@@ -338,16 +357,23 @@ class AsioRequester {
 
             const auto& properties = request->Properties();
 
-            std::string service =
-                request->Port().value_or(request->Https() ? "https" : "http");
+            std::string tcp_host =
+                request->HttpProxyHost().value_or(request->Host());
+
+            std::string tcp_port = request->HttpProxyPort().value_or(
```

Comment:
This could instead be:
> request->Https() ? "https" : "http"

since asio accepts "service names". A bit more abstract.